contracts/contracts/strategies/BaseCurveStrategy.sol

```diff
@@ -164,6 +140,58 @@ abstract contract BaseCurveStrategy is InitializableAbstractStrategy {
         IERC20(_asset).safeTransfer(_recipient, _amount);
     }
 
+    /**
+     * @dev Calculate amount of LP required when withdrawing specific amount of one
+     * of the underlying assets accounting for fees and slippage.
+     *
+     * Curve pools unfortunately do not contain a calculation function for
+     * amount of LP required when withdrawing a specific amount of one of the
+     * underlying tokens and also accounting for fees (Curve's calc_token_amount
+     * does account for slippage but not fees).
+     *
+     * Steps taken to calculate the metric:
+     *  - get amount of LP required if fees wouldn't apply
+     *  - increase the LP amount as if fees would apply to the entirety of the underlying
+     *    asset withdrawal. (when withdrawing only one coin fees apply only to amounts
+     *    of other assets pool would return in case of balanced removal - since those need
+     *    to be swapped for the single underlying asset being withdrawn)
+     *  - get amount of underlying asset withdrawn (this Curve function does consider slippage
+     *    and fees) when using the increased LP amount. As LP amount is slightly over-increased
+     *    so is amount of underlying assets returned.
+     *  - since we know exactly how much asset we require take the rate of LP required for asset
+     *    withdrawn to get the exact amount of LP.
+     */
+    function _calcCurveTokenAmount(uint256 _coinIndex, uint256 _amount)
+        internal
+        returns (uint256 required3Crv)
+    {
+        ICurvePool curvePool = ICurvePool(platformAddress);
+
+        uint256[3] memory _amounts = [uint256(0), uint256(0), uint256(0)];
+        _amounts[_coinIndex] = _amount;
+
+        // LP required when removing required asset ignoring fees
+        uint256 lpRequiredNoFees = curvePool.calc_token_amount(_amounts, false);
+        // LP required if fees would apply to entirety of removed amount
+        uint256 lpRequiredFullFees = lpRequiredNoFees.mulTruncateScale(
+            1e10 + curvePool.fee(),
+            1e10
+        );
+
+        /* asset received when withdrawing full fee applicable LP accounting for
+         * slippage and fees
+         */
+        uint256 assetReceivedForFullLPFees = curvePool.calc_withdraw_one_coin(
+            lpRequiredFullFees,
+            int128(uint128(_coinIndex))
```

Comment:
Any reason for not doing `int128(_coinIndex)` directly?

Reply:
seems to be solidity compiler limitation: 
```
TypeError: Explicit type conversion not allowed from "uint256" to "int128".
```

Reply:
I tried this on a solidity playground v0.4.24 and it worked, let me try it in 0.8.0"
```
  function hello(uint256 x) pure public returns (int128) {
    return int128(x);
  }
```

Reply:
hmm interesting...

Reply:
I get this on 0.8.7 compiler version

Reply:
Can confirm, Works on 0.4.24 but not on ^0.8.0. I guess it was changed for sanity reasons

Reply:
good point yeah probably